x/projects/keeper/msg_server_add_project_keys.go

```diff
@@ -21,24 +21,26 @@ func (k msgServer) AddProjectKeys(goCtx context.Context, msg *types.MsgAddProjec
 	}
 
 	// check if the admin key is valid
-	if !project.IsKeyType(adminKey, types.ProjectKey_ADMIN) || project.Subscription != adminKey {
+	if !project.IsKeyType(adminKey, types.ProjectKey_ADMIN) && project.Subscription != adminKey {
```

Comment:
why switch the OR to AND? this way, if someone maliciously creates a developer key with the same address as the admin, this check will pass